include/taskolib/exceptions.h

```diff
@@ -102,6 +102,79 @@ class Error : public std::runtime_error
     OptionalStepIndex index_;
 };
 
-} // namespace task
+} // namespace git
+
+namespace git {
+
+/**
+ * An exception class carrying an error message and, optionally, the index of the step in
```

Comment:
It does not make much sense for a Git error message to carry information about step indices. I guess we should just store the string and be done with it.